protocol.md

```diff
@@ -115,41 +120,51 @@ Except for the first *device*, which is validated in a specific way previously d
-4. *Tanker Core* shares the [Resource Encryption Key] with the recipients as described in [Sharing with users](#sharing-with-users) and [Sharing with user groups](#sharing-with-user-groups)
+1. *Tanker Core* generates the [Resource Encryption Key]
+2. *Tanker Core* symmetrically encrypts the given *data* with the [Resource Encryption Key]
+3. *Tanker Core* shares the [Resource Encryption Key] with the recipients as described in [Sharing with users](#sharing-with-users) and [Sharing with user groups](#sharing-with-user-groups)
 
 ### Sharing with users
 
 Prerequisite: the *user*'s *device* is authenticated against the *Tanker server*, and some *data* has been encrypted.
 
 Given the [Resource Encryption Key], sharing encrypted *data* with another *user* is done as follows:
 
-1. The *application* fetches the [Public Permanent Identity] for the recipient from the *application server*
-2. The *application* calls `tanker.share` with the [Public Permanent Identity]
-3. *Tanker Core* fetches the recipient's `device_creation` *block*s from the *Trustchain*
-4. *Tanker Core* verifies them and extract the recipient's public [User Encryption Key Pair]
-5. *Tanker Core* asymmetrically encrypts the [Resource Encryption Key] with the public [User Encryption Key Pair], creating the [Shared Encrypted Key]
-6. *Tanker Core* creates a `key_publish` *block* containing the [Shared Encrypted Key] and the recipient's public [User Encryption Key Pair], and pushes it to the *Trustchain*
-7. The *Tanker server* validates the *block* and notifies all the recipients' *device*s
-8. The recipient's *device* retrieves the *block*, verifies it and decrypts the [Shared Encrypted Key] using the private [User Encryption Key Pair], obtaining the [Resource Encryption Key]
-9. The *application* calls `tanker.decrypt` on the recipient's *device*
-10. *Tanker Core* decrypts the data using the [Resource Encryption Key]
+1. The *application* fetches the recipients' [Public Permanent Identity] from the *application server*
+1. *Tanker Core* fetches the recipients' `device_creation` *block*s from the *Trustchain*
+1. *Tanker Core* verifies the received `device_creation` *block*s
+1. *Tanker Core* extracts the recipient's public [User Encryption Key Pair]
+1. For each recipient, *Tanker Core* asymmetrically encrypts the [Resource Encryption Key] with the public recipient's [User Encryption Key Pair] creating a [Shared Encrypted Key] for each one of them
+1. For each recipient, *Tanker Core* creates a `key_publish` *block* containing the [Shared Encrypted Key] and the recipient's public [User Encryption Key Pair], and pushes it to the *Trustchain*
+1. The *Tanker server* validates the *block*
+
+## Data decryption
+
+Prerequisite: the user's has access to the encrypted data
+
+1. *Tanker Core* extracts the [Resource ID] from the encrypted data
+1. *Tanker Core* retrieves for the recipients' *device* the `key_publish` *block*,
+1. *Tanker Core* verifies the retrieved block
```

Comment:
not anymore, we should document that btw in block verification, and maybe say again here that we do *not* verify these blocks